packages/botonic-plugin-contentful/src/markup/markdown.ts

```diff
@@ -0,0 +1,89 @@
+import * as marked from 'marked'
+import { MarkUp, MarkupType, Token, TokenType } from './markup'
+
+/**
+ * marked's intex.ts does not include these types (though they work): del, strong, em
+ */
+export class Markdown extends MarkUp {
+  // https://marked.js.org/#/README.md#specifications
+  options: marked.MarkedOptions
+
+  constructor(flavour: MarkupType) {
+    super(flavour)
+    switch (flavour) {
+      case MarkupType.MARKDOWN:
+        this.options = {}
+        break
+      case MarkupType.GITHUB:
+      case MarkupType.CONTENTFUL:
+        this.options = { gfm: true }
+        break
+      default:
+        throw Error(`Invalid markdown flavour: ${flavour}`)
+    }
+  }
+
+  parse(txt: string): Token[] {
+    // cannot cache Lexer because it would accumulate the tokens
+    return marked.lexer(txt) as Token[]
+  }
+
+  renderToken(token: Token): string {
+    throw new Error('Not implemented')
+    // if (token.items) {
+    //   return this.render(token.items, '\n')
+    // }
+    // if (token.tokens && token.tokens.length > 1) {
+    //   return this.render(token.tokens, '')
+    // }
+    // return token.raw || token.text || ''
+  }
+  wrapWithInline(input: string, inlineType: TokenType): string {
+    if (inlineType === TokenType.EMPHASIS) {
+      return `_${input}_`
+    }
+    if (inlineType === TokenType.STRONG) {
+      return `**${input}**`
+    }
+    throw new Error(`wrapWithInline does not support inline ${inlineType}`)
+  }
+}
+
+// /**
```

Comment:
There's a lot of code commented out. Are you planning to make it work eventually?

Reply:
removed, thanks